### PR TITLE
key: 2.10.0 -> 2.12.2

### PR DESCRIPTION
**Pull request by FliegendeWurst**
Update to latest version: https://keyproject.github.io/key-docs/changelog/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/logic/key/default.nix
Comment (tomodachi94) [inline pkgs/applications/science/logic/key/default.nix:35]: Nitpick: This is used nowhere except for `meta.mainProgram`.

```suggestion
```

**Review by tomodachi94 — APPROVED** [review]
Lgtm, except for one nitpick you're free to ignore. In particular, thank you for noticing that upstream uses a newer Gradle!

**Review by tomodachi94 — APPROVED** [review]
As mentioned before, lgtm 💖 I'll merge this after Borgo passes.

**Review by fgaz — APPROVED** [review]
Thanks!

**Pull request merged**
